src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/ServerNode.cs

```diff
@@ -47,10 +47,19 @@ public ServerNode(ConnectionCompleteParams connInfo, IMultiServiceProvider servi
 
             this.context = new Lazy<SmoQueryContext>(() => CreateContext(serviceProvider));
 
-            NodeValue = connectionSummary.ServerName;
+            if (IsMasterConnection())

```

Comment:
I'm pretty sure there's something wrong with this approach. this root code is not really database instance node and there are some logic in the database node needed for expanding and scripting. The context for server node and database node are different and database has a smo object because it is scriptable. the tests should verify that